### PR TITLE
conservation projects in project list

### PR DESCRIPTION
**Pull request by rishsingh07**
Different marker colour for projects having conservation projects. 
Not sure about the marker colour I chose is desired one. Please help with that.

Changed all projects endpoint to - `/app/projects?filter[purpose]=trees,conservation&_scope=map&currency=${currency}` for conservation projects

**Todo -** 
- [x] redirect donation for conservation projects to planet donations
- [x] Donation URL for Conservation projects is yet to be implemented [Link eg:](https://donate-with-planet-a946yld87-planetapp.vercel.app/?to=conservation-wea-kenya&callback_url=https://planet-webapp-5dh39wlfk-planetapp.vercel.app/&country=DE&currency=EUR&locale=en&tenant=ten_NxJq55pm)
- [ ] ~Project Classification for conservation needs to be implemented~ can be merged without this

### How to Test -

- Open Latest Deployment
- Look for blue color markers on the map
- hover on the marker
- look for unit cost, verify that `amount per m2` is shown instead of `amount per tree`
- click on donate
- verify that unit cost on planet donations is the same as it was on webapp
- complete donation

### CODE REVIEW

**Comment by sagararyal** [comment]
#### 📝 Changed routes:


| [**/**](https://planet-webapp-i3y4dum5j-planetapp.vercel.app/) |
|:-:|
|<a href="https://planet-webapp-i3y4dum5j-planetapp.vercel.app/">Screenshot of /</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/BGvH2fy2HKA1xRRwmj6tfsirdAc17JnlfsSyGWZMwp-wKKXYyOoRtwBfeMXYZQyCdxqmxynkA3Hjlkg1z55VBw.png">(view full size)</a> |







Commit 2835fdacd0f859f7bef09ef37cda2c14cbbe5ffa (https://planet-webapp-i3y4dum5j-planetapp.vercel.app).

**Comment by norbertschuler** [comment]
- On staging the conversation projects are missing images and two of them has an undefined donation price:
 
Bildschirmfoto 2021-11-21 um 14 58 46

**Comment by norbertschuler** [comment]
- [ ] The donation form for the one conservation project having a donation amount reveals a design issue if the projects description text is too long (the error is as the correct API call is still work in progress):

Bildschirmfoto 2021-11-21 um 15 03 09

**Comment by sarveshpro** [comment]
> * On staging the conversation projects are missing images and two of them have an undefined donation price:
This is now fixed and production is updated with the latest data.

**Comment by sarveshpro** [comment]
> * [ ]  The donation form for the one conservation project having a donation amount reveals a design issue if the projects description text is too long (the validation error is probably caused as the correct API call to https://app-staging.plant-for-the-planet.org/app/donations is still work in progress):

This is not needed as it was decided to redirect user to donate.plant-for-the-planet.org for donations if project purpose is conservation

File: src/features/projects/components/ProjectSnippet.tsx
Comment (norbertschuler) [inline src/features/projects/components/ProjectSnippet.tsx:53]: If this should be a real full feature donation link, you should do it like it is done in https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/1363 (better wait for that PR to be merged, so you do not have to write the code twice and created merge conflicts).

File: src/features/projects/components/PopupProject.tsx
Comment (norbertschuler) [inline src/features/projects/components/PopupProject.tsx:44]: Same as https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/1318/files#r773669034

File: src/features/projects/components/ProjectSnippet.tsx
Comment (norbertschuler) [inline src/features/projects/components/ProjectSnippet.tsx:56]: Remove all console.log before merging.

**Comment by norbertschuler** [comment]
- [ ] Here is a project with only conservation projects showing no projects as conversation projects are not loaded with the API call https://app-staging.plant-for-the-planet.org/app/profiles/prf_DPuHSu4G1RXXIZ1nAvqG68CE/projects  and displayed here: https://planet-webapp-6ovz5gkzv-planetapp.vercel.app/t/womens-earth-alliance - should conservation projects also be shown here? I think this is a backend issue or is there already a way to do this API call with projects filters?

File: src/features/projects/components/ProjectSnippet.tsx
Comment (sarveshpro) [reply src/features/projects/components/ProjectSnippet.tsx]: That's merged in this now

**Comment by sarveshpro** [comment]
> * [ ]  Here is a project with only conservation projects showing no projects as conversation projects are not loaded with the API call https://app-staging.plant-for-the-planet.org/app/profiles/prf_DPuHSu4G1RXXIZ1nAvqG68CE/projects  and displayed here: https://planet-webapp-6ovz5gkzv-planetapp.vercel.app/t/womens-earth-alliance - should conservation projects also be shown here? I think this is a backend issue or is there already a way to do [this API call](https://plant-for-the-planet.stoplight.io/docs/treecounter-platform/b3A6Mjc4MjMwMg-retrieve-user-s-projects) with projects filters?

I had a chat with @kewalgangar and this will be a separate fix, Kewal will provide a different scope or versioned API which will include other project types in the TPO profile. For now, we just show it on the public map and donations only.

File: public/static/locales/en/donate.json
Comment (tejassonar) [inline public/static/locales/en/donate.json:115]: Maybe we should change the unit to m² here as well

**Comment by tejassonar** [comment]
Looks good to me except for the small unit change.

**Comment by norbertschuler** [comment]
- [ ] I made a test donation on staging, but my donation receipt shows an error when I try to download it with this link from my profile area: https://app-staging.plant-for-the-planet.org/certificate/U4EHMUU36EJL (`TCPDF ERROR: [Image] Unable to get the size of the image: /tmp/certXaJHAT `) - this should not happen on production. The related donation is https://app-staging.plant-for-the-planet.org/backend/donationContribution/574887/show

Bildschirmfoto 2022-01-15 um 11 53 37

File: public/static/locales/en/donate.json
Comment (norbertschuler) [reply public/static/locales/en/donate.json]: First I tried to do this with `<sup>2</sup>` in the translation resource, but then I thought we can also just enter the unicode representation `m²` into the resources.

File: src/features/projects/components/ProjectSnippet.tsx
Comment (norbertschuler) [inline src/features/projects/components/ProjectSnippet.tsx:138]: `.treeCosts` is still widely used within our code, e.g. at line 118 above this or in `PopupProject.tsx` - so I do not understand why we want to change it only right here now? For me the code get inconsistent this way and I won't change it in this PR if there is no reason to do so.

**Review by norbertschuler — DISMISSED** [review]

File: src/features/projects/components/ProjectSnippet.tsx
Comment (sarveshpro) [reply src/features/projects/components/ProjectSnippet.tsx:138]: replaced treeCost with unitCost in remaining places.

**Comment by tejassonar** [comment]
Clicking here 
image
Results into blank screen

**Review by sarveshpro — APPROVED** [review]
LGFM ✅

**Pull request merged**
